Automattic/wp-calypso
# Optimize Dashicons, Noticons, and Google fonts loading

This pull request seeks to optimize the loading of Dashicons, Noticons, and `Noto Serif` Google fonts, saving around 30K (gzipped) per page as well as some unnecessary requests.

More specifically, it defers the loading of Dashicons until they are actually used, i.e. when the user navigates to a page with the post editor (note that we are in the process of phasing out Dashicons and this is the only remaining place where it is still used). This was achieved by moving the CSS used by the `post-editor` section into its own bundle, thus reducing the size of the default `style.css` bundle.

It also makes sure Noticons and Dashicons are always loaded with the same cache number (i.e. `?v=20150727`) to take advantage of caching (and avoid loading the exact same content twice).

Finally, it updates our code to always load the `Noto Serif` Google font via HTTPS. Using the same protocol everywhere prevents the case where [two](http://fonts.googleapis.com/css?family=Noto+Serif:400,400i,700,700i&subset=cyrillic,cyrillic-ext,greek,greek-ext,latin-ext,vietnamese) [distinct](https://fonts.googleapis.com/css?family=Noto+Serif:400,400i,700,700i&subset=cyrillic,cyrillic-ext,greek,greek-ext,latin-ext,vietnamese) stylesheets would be loaded for the same font: despite being 99% similar they are actually protocol-specific (the stylesheet indeed contains urls which include the protocol used to load the stylesheet itself).

#### Testing instructions
 
1. Run `git checkout update/font-loading` and start your server, or open a [live branch](https://calypso.live/?branch=update/font-loading)
2. Clear all browsing data (cache, local data...)
3. Open the `Network` panel in your browser's devtools to analyze how file are loaded
4. Open the [`Login` page](http://calypso.localhost:3000/log-in)
5. Assert that the Dashicons stylesheet is no longer loaded
6. Assert that the Noticons stylesheet is loaded via HTTPS, with a cache number
7. Assert that the `Noto Serif` stylesheet is loaded via HTTPS
8. Open the [`404 Error` page](http://calypso.localhost:3000/404)
9. Assert that none of these stylesheets is loaded again (they should be in the cache)
10. Log in to WordPress.com
11. Navigate to any page with the post editor
12. Assert that the editor displays exactly as before
13. Assert that the Dashicons stylesheet is loaded once via HTTPS
14. Browse any page from Calypso
15. Assert that none of the aforementioned stylesheets is loaded again
 
#### Reviews
